public/app/plugins/datasource/tempo/SearchTraceQLEditor/GroupByField.tsx

```diff
@@ -0,0 +1,141 @@
+
+import { TraceqlFilter, TraceqlSearchScope } from '../dataquery.gen';
+import { TempoDatasource } from '../datasource';
+import { TempoQuery } from '../types';
+
+import InlineSearchField from './InlineSearchField';
+import { replaceAt } from './utils';
+
+interface Props {
+  datasource: TempoDatasource;
+  onChange: (value: TempoQuery) => void;
+  query: Partial<TempoQuery> & TempoQuery;
+}
+
+export const GroupByField = (props: Props) => {
+  const { datasource, onChange, query } = props;
+  const styles = useStyles2(getStyles);
+  const generateId = () => uuidv4().slice(0, 8);
+  const [isTagsLoading, setIsTagsLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchTags = async () => {
+      try {
+        await datasource.languageProvider.start();
+        setIsTagsLoading(false);
+      } catch (error) {
+        if (error instanceof Error) {
+          dispatch(notifyApp(createErrorNotification('Error', error)));
+        }
+      }
+    };
+    fetchTags();
+  }, [datasource]);
+
+  useEffect(() => {
+    if (!query.groupBy || query.groupBy.length === 0) {
+      onChange({
+        ...query,
+        groupBy: [
+          {
+            id: generateId(),
+            scope: TraceqlSearchScope.Span,
+          },
+        ],
+      });
+    }
+  }, [onChange, query]);
+
+  const getTags = (f: TraceqlFilter) => {
+    return datasource!.languageProvider.getMetricsSummaryTags(f.scope);
+  };
+
+  const addFilter = () => {
+    updateFilter({
+      id: generateId(),
+      scope: TraceqlSearchScope.Span,
+    });
+  };
+
+  const removeFilter = (filter: TraceqlFilter) => {
+    const copy = { ...query };
+    onChange({ ...copy, groupBy: copy.groupBy?.filter((f) => f.id !== filter.id) });
+  };
+
+  const updateFilter = (filter: TraceqlFilter) => {
+    const copy = { ...query };
+    copy.groupBy ||= [];
+    const indexOfFilter = copy.groupBy.findIndex((f) => f.id === filter.id);
+    if (indexOfFilter >= 0) {
+      // update in place if the filter already exists, for consistency and to avoid UI bugs
+      copy.groupBy = replaceAt(copy.groupBy, indexOfFilter, filter);
+    } else {
+      copy.groupBy.push(filter);
+    }
+    onChange(copy);
+  };
+
+  const scopeOptions = Object.values(TraceqlSearchScope).map((t) => ({ label: t, value: t }));
+
+  return (
+    <InlineSearchField
+      label="Group By"
+      tooltip="Select a tag to see the metrics summary. Note: the metrics summary API only considers spans of kind = server."
+    >
+      <>
+        {query.groupBy?.map((f, i) => (
+          <div key={f.id}>
+            <HorizontalGroup spacing={'none'} width={'auto'}>
+              <Select
+                options={scopeOptions}
+                value={f.scope}
+                onChange={(v) => {
+                  updateFilter({ ...f, scope: v?.value });
+                }}
+                placeholder="Select scope"
+                aria-label={`Select scope for ${f.id}`}
```

Comment:
if `id` is a generated UID, I don't think it would be of much help here?

Reply:
It's to make a distinction between multiple group by filters. Not sure what else to use here that would make it clear that this is a unique filter and different from the other group by filters.

Reply:
So I mean this is mainly for screen readers, right? So if this reads `Select scope for dJ2Az1` or something, I don't think that helps the user that much.

Reply:
I agree it would be better to have something that makes more sense for screen readers, but without the id here, every group by filter will just read as `Select scope`. To simplify this and to make sure the screen reader user knows which filter is which, I've just added the filter index here as opposed to its id.